okta/services/idaas/resource_okta_app_user_custom_schema_property.go

```diff
@@ -238,9 +238,9 @@ func updateAppUserSubSchemaProperty(ctx context.Context, d *schema.ResourceData,
 	retypeUserSchemaPropertyEnums(custom)
 	boc := utils.NewExponentialBackOffWithContext(ctx, 10*time.Second)
 	err = backoff.Retry(func() error {
-		_, resp, err := getOktaClientFromMetadata(meta).UserSchema.
-			UpdateApplicationUserProfile(ctx, d.Get("app_id").(string), *custom)
-		if utils.SuppressErrorOn404(resp, err) == nil {
+		_, resp, err := getOktaClientFromMetadata(meta).UserSchema.UpdateApplicationUserProfile(ctx, d.Get("app_id").(string), *custom)
+		// if error is nil, we skip this check entirely since utils.SuppressErrorOn404 will return nil either when err is nil or when resp.StatusCode is 404.
+		if err != nil && utils.SuppressErrorOn404(resp, err) == nil {
```

Comment:
```suggestion
		if err != nil || utils.SuppressErrorOn404(resp, err) == nil {
```
Should this be an `OR` rather than an `AND`

reasoning
```go
if err != nil && utils.SuppressErrorOn404(resp, err) == nil {
  // this code can never be reached
}

if err != nil || utils.SuppressErrorOn404(resp, err) == nil {
  // this code can be reached if either:
  // 1. err != nil
  // 2. utils.SuppressErrorOn404(resp, err) == nil
}
```